File: docs/2014/snippets/csharp/SQL14/integration services runtime api/application/cs/application_collections.cs
--<snippetApplication_DataTypeInfosCS>
class ApplicationTests1
    {
        static void Main(string[] args)
        {
            DataTypeInfos dti = new Application().DataTypeInfos;

            foreach (DataTypeInfo x in dti) 
                Console.WriteLine(x.TypeName + " , " + x.TypeEnumName);
        }
    }
--</snippetApplication_DataTypeInfosCS>
--<snippetApplication_DBProviderInfosCS>
class ApplicationTests
    {
        static void Main(string[] args)
        {
            Application app = new Application();
            DBProviderInfos dbis = app.DBProviderInfos;
            Console.WriteLine("Database Provider Info:");
            foreach (DBProviderInfo dbi in dbis)
                Console.WriteLine(dbi.Description);
        }
    }
--</snippetApplication_DBProviderInfosCS>
--<snippetApplication_ForEachEnumeratorInfosCS>
class ApplicationTests
    {
        static void Main(string[] args)
        {
            Application app = new Application();
            ForEachEnumeratorInfos fee = app.ForEachEnumeratorInfos;
            foreach(ForEachEnumeratorInfo x in fee)
                Console.WriteLine(x.ID);
        }
    }
--</snippetApplication_ForEachEnumeratorInfosCS>
--<snippetApplication_LogProviderInfosCS>
class ApplicationTests
    {
        static void Main(string[] args)
        {
            Application app = new Application();
            LogProviderInfos lpis = app.LogProviderInfos;
            foreach (LogProviderInfo x in lpis)
                Console.WriteLine(x.Name);
        }
    }
--</snippetApplication_LogProviderInfosCS>
--<snippetApplication_PipelineComponentInfosCS>
class ApplicationTests
    {
        static void Main(string[] args)
        {
            Application app = new Application();
            PipelineComponentInfos pcis = app.PipelineComponentInfos;
            foreach (PipelineComponentInfo x in pcis)
                Console.WriteLine(x.CreationName);
        }
    }
--</snippetApplication_PipelineComponentInfosCS>
--<snippetApplication_TaskInfosCS>
class ApplicationTests
    {
        static void Main(string[] args)
        {
            Application app = new Application();

            TaskInfos tis = app.TaskInfos;
            foreach (TaskInfo x in tis)
                Console.WriteLine(x.FileName + "..." + x.Description);
        }
    }
--</snippetApplication_TaskInfosCS>

File: docs/2014/snippets/visualbasic/SQL15/integration services runtime api/application/vb/application_collections.vb
' <snippetApplication_DataTypeInfosVB>
class ApplicationTests1
    Sub Main(ByVal args() As String)
        Dim dti As DataTypeInfos =  New Application().DataTypeInfos 
 
        For Each x As DataTypeInfo In dti
            Console.WriteLine(x.TypeName + " , " + x.TypeEnumName)
        Next
    End Sub
End Class
' </snippetApplication_DataTypeInfosVB>
' <snippetApplication_DBProviderInfosVB>
class ApplicationTests
    Sub Main(ByVal args() As String)
        Dim app As Application =  New Application() 
        Dim dbis As DBProviderInfos =  app.DBProviderInfos 
        Console.WriteLine("Database Provider Info:")
        For Each dbi As DBProviderInfo In dbis
            Console.WriteLine(dbi.Description)
        Next
    End Sub
End Class
' </snippetApplication_DBProviderInfosVB>
' <snippetApplication_ForEachEnumeratorInfosVB>
class ApplicationTests
    Sub Main(ByVal args() As String)
        Dim app As Application =  New Application() 
        Dim fee As ForEachEnumeratorInfos =  app.ForEachEnumeratorInfos 
        For Each x As ForEachEnumeratorInfo In fee
            Console.WriteLine(x.ID)
        Next
    End Sub
End Class
' </snippetApplication_ForEachEnumeratorInfosVB>
' <snippetApplication_LogProviderInfosVB>
class ApplicationTests
    Sub Main(ByVal args() As String)
        Dim app As Application =  New Application() 
        Dim lpis As LogProviderInfos =  app.LogProviderInfos 
        For Each x As LogProviderInfo In lpis
            Console.WriteLine(x.Name)
        Next
    End Sub
End Class
' </snippetApplication_LogProviderInfosVB>
' <snippetApplication_PipelineComponentInfosVB>
class ApplicationTests
    Sub Main(ByVal args() As String)
        Dim app As Application =  New Application() 
        Dim pcis As PipelineComponentInfos =  app.PipelineComponentInfos 
        For Each x As PipelineComponentInfo In pcis
            Console.WriteLine(x.CreationName)
        Next
    End Sub
End Class
' </snippetApplication_PipelineComponentInfosVB>
' <snippetApplication_TaskInfosVB>
class ApplicationTests
    Sub Main(ByVal args() As String)
        Dim app As Application =  New Application() 
        Dim tis As TaskInfos =  app.TaskInfos 
        For Each x As TaskInfo In tis
            Console.WriteLine(x.FileName + "..." + x.Description)
        Next
    End Sub
End Class
' </snippetApplication_TaskInfosVB>




